File: homecloud_core/defaults.py
"""Platform endpoint defaults — hidden from end users."""

from __future__ import annotations

from homecloud_core.env import env_apex

DEFAULT_APEX = "holab.abrdns.com"
DEFAULT_PROFILE = "default"
WHOAMI_PATH = "/access-key/whoami"
WHOAMI_ACCOUNT_SENTINEL = "-"


def platform_apex() -> str:
    return env_apex() or DEFAULT_APEX


def console_url(apex: str | None = None) -> str:
    host = apex or platform_apex()
    return f"https://console.{host}/api/v1"


def console_web_url(apex: str | None = None) -> str:
    host = apex or platform_apex()
    return f"https://console.{host}"


def mq_url(apex: str | None = None) -> str:
    host = apex or platform_apex()
    return f"https://mq.{host}"


def so_url(apex: str | None = None) -> str:
    host = apex or platform_apex()
    return f"https://so.{host}"


def secrets_url(apex: str | None = None) -> str:
    host = apex or platform_apex()
    return f"https://secrets.{host}"


def mail_api_url(apex: str | None = None) -> str:
    """Mail data-plane SigV1 host (not SMTP mail.{domain})."""
    host = apex or platform_apex()
    return f"https://mailapi.{host}"


def function_url(name: str, apex: str | None = None) -> str:
    """Data-plane Function URL: https://{name}.func.{apex}."""
    host = apex or platform_apex()
    return f"https://{name.strip().lower()}.func.{host}"

File: js/src/defaults.js
"use strict";

const DEFAULT_APEX = "holab.abrdns.com";
const DEFAULT_PROFILE = "default";
const WHOAMI_PATH = "/access-key/whoami";
const WHOAMI_ACCOUNT_SENTINEL = "-";

function soUrl(apex) {
  return `https://so.${apex || DEFAULT_APEX}`;
}
function mqUrl(apex) {
  return `https://mq.${apex || DEFAULT_APEX}`;
}
function secretsUrl(apex) {
  return `https://secrets.${apex || DEFAULT_APEX}`;
}
function mailApiUrl(apex) {
  return `https://mailapi.${apex || DEFAULT_APEX}`;
}
function consoleUrl(apex) {
  return `https://console.${apex || DEFAULT_APEX}/api/v1`;
}
function functionUrl(name, apex) {
  return `https://${String(name).trim().toLowerCase()}.func.${apex || DEFAULT_APEX}`;
}

function envFirst(...names) {
  for (const name of names) {
    const v = process.env[name];
    if (v != null && String(v).trim() !== "") return v;
  }
  return null;
}

module.exports = {
  DEFAULT_APEX,
  DEFAULT_PROFILE,
  WHOAMI_PATH,
  WHOAMI_ACCOUNT_SENTINEL,
  soUrl,
  mqUrl,
  secretsUrl,
  mailApiUrl,
  consoleUrl,
  functionUrl,
  envFirst,
};
